src/App/Fossa/Config/Report.hs

```diff
@@ -136,11 +157,13 @@ mergeOpts cfgfile envvars ReportCliOptions{..} = do
     <*> pure cliReportType
     <*> revision
 
-validateOutputFormat :: Has Diagnostics sig m => Bool -> m ReportOutputFormat
-validateOutputFormat doJson =
-  if doJson
-    then pure ReportJson
-    else fatalText "Plaintext reports are not available for this report."
+validateOutputFormat :: Has Diagnostics sig m => Bool -> Maybe String -> m ReportOutputFormat
+validateOutputFormat True _ = pure ReportJson
+validateOutputFormat False Nothing = fatalText "Plaintext reports are not available for this report."
+validateOutputFormat False (Just format) | format == show ReportJson = pure ReportJson
+validateOutputFormat False (Just format) | format == show ReportMarkdown = pure ReportMarkdown
+validateOutputFormat False (Just format) | format == show ReportSPDX = pure ReportSPDX
+validateOutputFormat False (Just format) = fatalText $ "Report format " <> toText format <> " is not supported"
```

Comment:
Three notes:
- We can simplify this by deriving `Read` on the `ReportOutputFormat` type, and using the `readMaybe` helper:
- We should use concrete Error types.
- The error for running without options should be updated to say something "You have to use either `--format foo` or `--json`".  Or we should find a good default, since not providing options should generally not cause failure.  This was a bad decision from when I first implemented reports in v2, I'm glad we're cleaning this up as a side-effect of SPDX reports.

```suggestion
validateOutputFormat :: Has Diagnostics sig m => Bool -> Maybe String -> m ReportOutputFormat
validateOutputFormat True _ = pure ReportJson
validateOutputFormat False Nothing = fatal NoFormatProvided
validateOutputFormat False (Just format) = fromMaybe (InvalidReportFormat format) $ readMaybe format
```

You'll have to add 
```haskell
data NoFormatProvided = NoFormatProvided
instance ToDiagnostic NoFormatProvided where
  something something = something $ "Plaintext reports are not available for this report."
```

Or whatever error message you think is best.

Reply:
ooh, thanks!

Reply:
Normally, using `Read` is verboten.  But in this case, I think it makes a lot of sense, and we're using the safe `readMaybe` as opposed to the unsafe `read`.

Reply:
> Normally, using Read is verboten. But in this case, I think it makes a lot of sense, and we're using the safe readMaybe as opposed to the unsafe read.

Does this mean that in reality, using `read` is verboten, not implementing `Read`?

(Also, deriving `Read` doesn't work for us, because it looks for e.g. `ReportJson`, not `json`, when derived- but I'll manually implement it.)

Reply:
Ah, I mixed up how `Show` and `Read` work together.  For some reason, I thought that deriving `Read` would change based on the impl of `Show`, but that's obviously not true.

Reply:
Updated!